Modal 컴포넌트 구현

Jeongeun-Choi opened the pull request:
## 📕 제목

Modal 컴포넌트 구현
관련이슈 #37

## 📗 작업 내용

> 구현 내용 및 작업 했던 내역

- [x] 배경색은 진한 회색으로 한다. 
- [x] 배경을 눌렀을때 모달창이 닫히도록 구현합니다.
- [x] 취소 버튼이 존재하면 취소 버튼을 눌러도 모달창이 닫히도록 구현합니다.

## 📘 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- Modal 컴포넌트를 사용하고싶을때
  - <strong>``<editor-modal type='source' title='소스 불러오기'></editor-modal>``</strong> 형식으로 불러옵니다.
![image](https://user-images.githubusercontent.com/55783203/100178390-1c457780-2f17-11eb-80ee-9f5715b9d78d.png)

- editor-modal 컴포넌트 내부 모습
![image](https://user-images.githubusercontent.com/55783203/100178447-37b08280-2f17-11eb-9074-26f4ceda9e45.png)

- modal/modalContents.js에 modal content로 사용할 컴포넌트를 지정해줍니다.
![image](https://user-images.githubusercontent.com/55783203/100178605-878f4980-2f17-11eb-83a8-b4c5f9cae19b.png)
  - modal에서 받은 type을 키로 설정해준다.
  - 키값은 modal 안에 넣을 컴포넌트를 넣어줍니다.
  - <strong>반드시 class=modal-component을 넣어줘야합니다!</strong> (그렇지 않으면 width 설정이 제대로 안먹히더라구요...)

- modal-buttons 컴포넌트
![image](https://user-images.githubusercontent.com/55783203/100178740-e81e8680-2f17-11eb-98ee-9d3c4ad932cc.png)

- modalButtons에 아래와 같이 UI prototype에 맞게 작성해주시면 됩니다! (파일 저장 부분은 source와 동일하고 저장하기로 문자열만 바꿔주면 됩니다! effect는 UI Prototype에서 확인 버튼이 없어 취소 버튼 하나만 나오도록 했습니다.)
![image](https://user-images.githubusercontent.com/55783203/100178785-fff60a80-2f17-11eb-8766-1a634ee31331.png)
  - 하드코딩으로 이루어져있으며 아토믹하게 만들 수도 있을 것 같아 추후에 리팩토링 작업을 할 것 같습니다😉

Jeongeun-Choi committed:
feat: [FE] Modal Close Event 구현

- 본문 내용
Modal 배경화면을 누르거나 취소 버튼을 눌렸을때 Modal 창이 닫히는 함수를
구현했습니다.

Jeongeun-Choi committed:
style: [FE] Modal Style 추가

- 본문 내용
 - Modal conent가 정중앙에 오도록 구현했습니다.
 - 자주 쓰일 것 같은 색상을 따로 변수로 두었습니다.

Jeongeun-Choi committed:
style: [FE] Modal-content에 display 추가

- 본문 내용
class 이름이 modal-content인 영역에 display를 추가하여 가운데 정렬을
하도록했습니다.

Jeongeun-Choi committed:
style: [FE] Modal buttons css 추가

- 본문 내용
Modal buttons css 내용 추가했습니다.

pieisland commented on frontend/src/components/modal/style.scss:
```diff
@@ -0,0 +1,62 @@
+$borderColor: #e0e0e0;
+$greenColor: #03c75a;
+
+%modal-buttons {
+  width: 90%;
+  display: flex;
+  justify-content: center;
+  margin-top: 1rem;
+}
```
scss 에서 %로 뭔가 할 수 있는 게 있나보네요. 처음 알았어요 ㅋㅋ 공부해야겠습니다.

Songwonseok reviewed (approved):
혹시 화살표 함수를 사용 안하고 this를 bind하신 이유가 있나요?

wooojini commented on frontend/src/components/modal/style.scss:
```diff
@@ -0,0 +1,62 @@
+.modal-content {
+  width: 50%;
+  height: 60%;
+  display: flex;
```
이전 리뷰에도 남겼는데 CSS 정렬을 해주면 어떨까요???
참조 : https://webdesign.tutsplus.com/articles/outside-in-ordering-css-properties-by-importance--cms-21685

pieisland commented on frontend/src/components/modal/style.scss:
```diff
@@ -0,0 +1,62 @@
+$baseColor: #212121;
+$borderColor: #e0e0e0;
+$greenColor: #03c75a;
```
변수 지정해서 색 쓰는 거 좋네요.

pieisland reviewed (approved):
모달 깔끔하게 만들어졌네요! 수고하셨습니다~

wooojini reviewed (approved):
고생하셨습니다!! 모달을 재사용하기 위해서 고민한 흔적들이 보이네요!!! 
 혹시 모달을 컴포넌트로 분리해주는 것은 어떨까요??? 감싸주어야하는 HTML 부분이 많아서 중간에 변경사항이 생긴다면 해당 부분의 코드를 모두 찾아서 변경해주어야 할 것 같습니다:-)

Jeongeun-Choi committed:
style: [FE] 중복된 border 삭제

- 본문 내용
중복된 boorder를 삭제했습니다.

Jeongeun-Choi committed:
style: [FE] css 정렬

- 본문 내용
css 정렬을 했습니다.

Jeongeun-Choi committed:
fix: [FE] Modal 컴포넌트 재사용성 고려하여 다시 작성

- 본문 내용
  - modal의 title과 type을 받을 수 있습니다.
  - modalContents에 받은 type을 넣어 줄 수 있습니다.
  - modal-buttons에도 type을 넣어줘야합니다.

Jeongeun-Choi committed:
feat: [FE] ModalContents Object 생성

- 본문 내용
  - modal에서 받은 type을 키로 설정해준다.
  - 키값은 modal 안에 넣을 컴포넌트를 넣어줍니다.

Jeongeun-Choi committed:
style: class name 추가

- 본문 내용
필요한 class name을 추가했습니다.

Jeongeun-Choi committed:
feat: [FE] Modal buttons 분리

- 본문 내용
  - 하드 코딩을 하여 각 모달의 type에 맞게 출력되는 버튼을 만들었습니다.

Jeongeun-Choi committed:
feat: attributeChangedCallback 함수 추가

- 본문 내용
attributeChangedCallback 함수를 추가했습니다.
